src/unity/Assets/Crest/Scripts/LodData/LodDataFlow.cs

```diff
@@ -1,38 +1,82 @@
 ﻿// This file is subject to the MIT License as seen in the root of this folder structure (LICENSE)
 
 using UnityEngine;
+using UnityEngine.Rendering;
 
 namespace Crest
 {
     /// <summary>
     /// A persistent flow simulation that moves around with a displacement LOD. The input is fully combined water surface shape.
     /// </summary>
-    public class LodDataFlow : LodDataPersistent
+    public class LodDataFlow : LodData
     {
         public override SimType LodDataType { get { return SimType.Flow; } }
-        protected override string ShaderSim { get { return "Ocean/Shape/Sim/Flow"; } }
         public override RenderTextureFormat TextureFormat { get { return RenderTextureFormat.RGHalf; } }
-        protected override Camera[] SimCameras { get { return OceanRenderer.Instance._camsFlow; } }
+        public override CameraClearFlags CamClearFlags { get { return CameraClearFlags.Color; } }
+        public override RenderTexture DataTexture { get { return Cam.targetTexture; } }
+        public override void UseSettings(SimSettingsBase settings) {}
+        public override SimSettingsBase CreateDefaultSettings() { return null;}
 
-        public override SimSettingsBase CreateDefaultSettings()
+        protected override void Start() {
+            base.Start();
+
+            _dataReadback = GetComponent<ReadbackLodData>();
+            _dataReadback._textureFormat = ReadbackLodData.TexFormat.RGHalf;
+            _dataReadback._active = true;
+        }
+
+        void OnDisable() {
+            // free native array when component removed or destroyed
+            if (_dataReadback._dataNative.IsCreated)
+            {
+                _dataReadback._dataNative.Dispose();
+            }
+        }
+
+        public static int SuggestDataLOD(Rect sampleAreaXZ, float minSpatialLength)
```

Comment:
hm wonder if duping this could be avoided. dont think its required for this PR though.

Reply:
The version in LodDataFlow is slightly different from the existing one. Just as a note.